gax/src/main/java/com/google/api/gax/rpc/ServerStreamingAttemptCallable.java

```diff
@@ -358,9 +361,14 @@ private void onAttemptResponse(ResponseT message) {
    * RetryingFuture} via the {@link #innerAttemptFuture}.
    */
   private void onAttemptError(Throwable throwable) {
-    if (cancellationCause != null) {
+    Throwable localCancellationCause;
+    synchronized (lock) {
```

Comment:
This looks right to me. But could it have any negative impact? E.g. introducing need to wait on the lock when previously that didn't happen?

Reply:
I should be ok (I also consulted with igorbernstein). The scope of the lock does not leave the `ServerStreamingAttemptCallable` class, this should guarantee no long lock waits.